File: sorted_squared_array/sorted_squared_array.py
# O(n log(n)) time O(n) space
def sorted_squared_array(array):
  squared_array = []

  for value in array:
    squared_array = squared_array + [value ** 2]

  squared_array.sort()
  return squared_array

print(sorted_squared_array([-2, -1, 1, 2, 3]))

File: sorted_squared_array/sorted_squared_array.java
import java.util.*;

public class Main {
    public static int[] sortedSquaredArray(int[] array) {
        int[] squaredArray = new int[array.length];
        for (int i = 0; i < array.length; i++) {
            squaredArray[i] = array[i] * array[i];
        }
        Arrays.sort(squaredArray);
        return squaredArray;
    }

    public static void main(String[] args) {
        int[] result = sortedSquaredArray(new int[]{-2, -1, 1, 2, 3});
        System.out.println(Arrays.toString(result)); // [1, 1, 4, 4, 9]
    }
}
